In [7]:
from sklearn.datasets import load_sample_images
import tensorflow as tf

import matplotlib.pyplot as plt

In [21]:
images = load_sample_images()["images"]
images = tf.keras.layers.CenterCrop(height=70, width=120)(images)
images = tf.keras.layers.Rescaling(scale = 1/255)(images)

In [28]:
conv_layer = tf.keras.layers.Conv2D(filters=32, kernel_size=7)
fmaps = conv_layer(images)

In [29]:
fmaps.shape

TensorShape([2, 64, 114, 32])

2 images, dimensions 64 x 114, with 32 "channels" which come from the filter size
we set above.

Note the change in pixels which got reduced from 70x120 due to the padding.
Kernel size is 7, so three pixels on each side get lost.

In [30]:
kernels, biases = conv_layer.get_weights()
kernels.shape

(7, 7, 3, 32)

The dimentions of the kernel is [kernel_height, kernel_width, input_channels, output_channels]. Biases is a 1-D array of size output_channels. 

Note the many hyperparameters for conv2D
- filters
- kernel size
- padding
- strides
- actiavtion function
- kernel initializer
- etc.

